### PR TITLE
Fixes #13234 - Turbolinks leading to certain pages can be turned off

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ohadlevy** [comment]
I'm not sure of the usage case, why can't the originator link simply add data-no-turbolink ?

**Comment by tstrachota** [comment]
@ohadlevy the idea behind is to let plugins disable turbolinks for pages that have been extended and need to load additional assets. This way they don't need to care about where are all the links to the page located.

**Comment by tstrachota** [comment]
@xprazak2 is there any chance of adding functional tests similar to what you did in https://github.com/theforeman/foreman/pull/3063 ? I would be good to the link_to itself as it's actually the core of this PR.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -32,6 +33,32 @@ def link_to(*args, &block)
     end
   end
 
+  def filter_turbolinks(args)
```
Comment (tstrachota) [inline app/helpers/application_helper.rb]: Watch out, `link_to` arguments are shifted when block is passed. See the implementation in http://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to I'd say, for simplicity's sake, it's ok to register just regex or string and match against `url_for` for the link that's being created.

**Comment by dLobatog** [comment]
@xprazak2 @tstrachota After reading the ticket, I didn't understand a few things:

```
This is required for pages that combine foreman UI with angular pagelets. 
```

Is there any example of that? How do turbolinks affect angular pagelets, and why `data-no-turbolink` as @ohadlevy doesn't work to link to other pages if needed? 

```
With turbolinks it doesn't load additional assets required by angular/bastion.
```

angular/bastion works well with Katello/Foreman + Turbolinks, what assets are not loading through pagelets, and why is it just pagelets that don't work with this?

Building and maintaining a whole service to keep track of disabling a Rails feature doesn't sound like should belong in Foreman, but I don't understand the whole issue yet. Do we have any pagelet that needs this, anywhere? I'll refrain to comment on the code issues until I understand what's exactly fixing, and how, as I don't want to cause @xprazak2 to change the code back and forth.

**Comment by tstrachota** [comment]
@dLobatog bastion works well with turbolinks because they actually disable them for links in the menu. See https://github.com/Katello/katello/blob/master/lib/katello/plugin.rb#L12 Turbolinks don't reload assets (that's why they are here for). So when you move from a standard foreman page with foreman assets to a katello one with turbolinks enabled, you don't get the additional bastion code and the page breaks.

An example of such pagelet is in this PR https://github.com/Katello/katello/pull/5710

I'm ok with just using `data-no-turbolink` on the smart proxy index page if we agree that's enough in this case, but I see several benefits of this PR's approach:
- no need to look for all links that direct to a page that needs to be loaded without turbolinks. When new link is added, plugins keep working.
- no need to disable turbolinks for that page in plain foreman installation
- we get clear entrypoint for plugins

**Comment by dLobatog** [comment]
@tstrachota If the problem is just not getting the bastion assets, you can add that wherever you need with deface as I posted on #3081 - https://github.com/theforeman/foreman/pull/3081/#discussion_r50032082

``` ruby
Deface::Override.new(
  :virtual_path => 'layouts/application', # or just the smart proxy in this case?
  :name => 'bastion_assets',
  :insert_after => 'erb[loud]:contains("layouts/base")',
  :text => "<%= render :file => 'bastion/layouts/assets' %>")
```

That way you wouldn't have to disable the turbolinks links, and pages load faster as the browser won't try to fetch assets on every new page.

**Comment by dLobatog** [comment]
Forgot to say, plugins can also use deface `add_to_attributes` to remove turbolinks links.

**Comment by xprazak2** [comment]
Closing as per discussion on IRC

**Comment by dLobatog** [comment]
Here's the aforementioned discussion - 
screenshot from 2016-01-20 14-50-43

**Pull request closed without merging**